module/VuFindTheme/src/VuFindTheme/View/Helper/AssetManager.php

```diff
@@ -0,0 +1,402 @@
+     * @param array  $attrs    Array of script attributes
+     * @param string $position Position to output script (header or footer)
+     *
+     * @return void
+     */
+    public function forcePrependScriptFile(
+        string $src,
+        string $type = 'text/javascript',
+        array $attrs = [],
+        string $position = 'header'
+    ): void {
+        $newScripts = [compact('src', 'type', 'attrs')];
+        foreach ($this->scripts[$position] as $script) {
+            if ($script['src'] ?? null !== $newScripts[0]['src']) {
+                $newScripts[] = $script;
+            }
+        }
+        $this->scripts[$position] = $newScripts;
+    }
+
+    /**
+     * Prepend raw script code.
+     *
+     * @param string $script              Script code
+     * @param string $type                Script type
+     * @param array  $attrs               Additional attributes for the script tag
+     * @param bool   $allowArbitraryAttrs Should we allow arbitrary attributes in $attrs?
+     * @param string $position            Position to output script (header or footer)
+     *
+     * @return void
+     */
+    public function prependScript(
+        string $script,
+        string $type = 'text/javascript',
+        array $attrs = [],
+        bool $allowArbitraryAttrs = false,
+        string $position = 'header'
+    ) {
+        array_unshift($this->scripts[$position], compact('script', 'type', 'attrs', 'allowArbitraryAttrs'));
+    }
+
+    /**
+     * Given a relative JS or CSS path, apply appropriate theme prefixing if possible; return null if
+     * the resource could not be found in a theme.
+     *
+     * @param string $relPath Relative path to find in theme
+     *
+     * @return ?string
+     */
+    protected function applyThemeToRelativePath(string $relPath): ?string
+    {
+        $details = $this->themeInfo->findContainingTheme($relPath, ThemeInfo::RETURN_ALL_DETAILS);
+        if (!empty($details)) {
+            $urlHelper = $this->getView()->plugin('url');
+            $url = $urlHelper('home') . "themes/{$details['theme']}/" . $relPath;
+            $url .= strstr($url, '?') ? '&_=' : '?_=';
+            $url .= filemtime($details['path']);
+            return $url;
+        }
+        // Cannot find in theme? Return null.
+        return null;
+    }
+
+    /**
+     * Return the HTML to output script assets in the requested position.
+     *
+     * @param mixed $position Position of assets (header or footer)
+     *
+     * @return string
+     */
+    protected function outputScriptAssets($position): string
+    {
+        $output = [];
+        $scriptHelper = $this->getView()->plugin('inlineScript');
+        $processedScripts = $this->pipeline->process($this->scripts[$position], 'js');
+        foreach ($processedScripts as $i => $script) {
+            if ($script['allowArbitraryAttrs'] ?? false) {
+                $scriptHelper->setAllowArbitraryAttributes(true);
```

Comment:
Should this be reset afterwards?

Reply:
Definitely! Fixed by 3e1d4eeaf900e3725dfe8a82060612b757ef09dc.

Reply:
While working on unit tests, I realized that there were some problems and inconsistencies here; addressed by 6d0231de0be1778de4ae34d926cd31ba82b67d34.